### ISSUE TITLE
Allow configuring sidecarContainers for fluentd pod

### ISSUE THREAD

**Issue by abierbaum** [kube-logging/logging-operator#1064]
**Is your feature request related to a problem? Please describe.**

I have fluentd running with an multiple out_file outputs to collect logs centrally in an AWS EKS cluster.  They collect the files into a mounted volume backed by elastic block storage.  I need to rotate the logs files to make sure they don't exceed the storage size of the ESB.  

To help support this I started thinking it would be nice to be able to add sidecar containers to the fluentd pod created by logging_operator.  I did not find a clean way to do this.

**Describe the solution you'd like**

I propose extending the logging_operator Fluentd section of the Logging CRD to support a sidecarContainers similar to how prometheus does in their helm chart.  There is a good discussion of it here: https://stackoverflow.com/questions/62909250/how-do-i-add-sidecar-containers-on-the-prometheus-stable-helm-charts-values-yaml/62910122#62910122

Something like:

```
sidecarContainers:
   - <container spec>
   - <container spec>
```

**Describe alternatives you've considered**

I considered doing a post deployment operation to patch all fluentd pods after the fact, but this doesn't integrate well with the logging operator since it could restart the fluentd pod.

**Additional context**

N/A

**Comment by pepov** [comment]
This could be a valid feature request for a future major release, thanks for reporting! (and sorry for the late reply)

**Comment by pepov** [comment]
We can reconsider adding this capability to the fluentd spec as it won't significantly increase the size of the CRD (905->931K) and we will eventually remove the fluentbitSpec and nodeAgents which will help reduce it to 750k.

**Comment by pepov** [comment]
closed with #1592
